### PR TITLE
minitube: update to 3.0

### PR DESCRIPTION
**Pull request by ArmedAviator**
The upstream maintainer is using Git submodules for parts of the release which are not included in the 3.0 tarball release.  I had asked him to release the submodules in the release or make tarball releases of each of the submodules to ensure every build is the same in XBPS.  Until then, this builds against the latest Git recursively.  A version 3.0_2 PR will be made if/when the upstream maintainer releases the tarballs for the submodules.

### CODE REVIEW

File: srcpkgs/minitube/template [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+short_desc="Qt5-based YouTube client"
 maintainer="Juan RP <xtraeme@voidlinux.org>"
-license="GPL-3"
+license="GPL-3-or-later"
 homepage="http://flavio.tordini.org/minitube"
-distfiles="https://github.com/flaviotordini/minitube/archive/$version.tar.gz"
-checksum=0f51fb86d0fdfbcf00af142868ceeacb6c27eb8a1e36d856162e23d63e90eb07
+
+do_fetch() {
+	git clone --recursive https://github.com/flaviotordini/minitube.git ${wrksrc}
```
Comment (maxice8) [inline srcpkgs/minitube/template]: i don't like it, can't the submodules be put in distfiles= ?

Comment (ArmedAviator) [reply srcpkgs/minitube/template]: I agree.  I'm not a fan.  The submodules do not have tarballs available.  I had asked the maintainer to make them via Github so the main package and 3 submodules can be downloaded and checksums verified.

https://github.com/flaviotordini/minitube/issues/133

**Comment by maxice8** [comment]
On Mon, Apr 15, 2019 at 11:45:10AM -0700, ArmedAviator wrote:
> ArmedAviator commented on this pull request.
> 
> 
> 
> >  homepage="http://flavio.tordini.org/minitube"
> -distfiles="https://github.com/flaviotordini/minitube/archive/$version.tar.gz"
> -checksum=0f51fb86d0fdfbcf00af142868ceeacb6c27eb8a1e36d856162e23d63e90eb07
> +
> +do_fetch() {
> +	git clone --recursive https://github.com/flaviotordini/minitube.git ${wrksrc}
> 
> I agree.  I'm not a fan.  The submodules do not have tarballs available.  I had asked the maintainer to make them via Github so the main package and 3 submodules can be downloaded and checksums verified.
> 
> https://github.com/flaviotordini/minitube/issues/133
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10994#discussion_r275495343

The best case is if upstream uploads their own tarball with the modules included

**Comment by maxice8** [comment]
Any luck ?

**Comment by ArmedAviator** [comment]
I and an Arch maintainer explained to him the problem and solution.  Give it a another day or so, I think he will be convinced to package upstream appropriately.  He appears to be in Spain so the back-and-forth is a slow.

**Comment by maxice8** [comment]
Ack

**Comment by ArmedAviator** [comment]
Looks like the upstream maintainer doesn't understand the situation so I don't think he'll change his mind any time soon.  How should we proceed?  Is there a way to have git clone up to a specific commit?

**Comment by maxice8** [comment]
On Fri, Apr 19, 2019 at 09:04:19AM -0700, ArmedAviator wrote:
> Looks like the upstream maintainer doesn't understand the situation so I don't think he'll change his mind any time soon.  How should we proceed?  Is there a way to have git clone up to a specific commit?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10994#issuecomment-484942541

bummer, feel free to fetch the submodules on the distfiles= and them copy them from ../${wrksrc}/project into ${wrksrc}/ .

**Comment by maxice8** [comment]
See the `obs` template for an example

**Comment by ArmedAviator** [comment]
Thanks for the tip.  Got it working with that method.  I'll keep an eye out for upstream to package it all together.

**Comment by maxice8** [comment]
On Fri, Apr 19, 2019 at 10:06:36AM -0700, ArmedAviator wrote:
> Thanks for the tip.  Got it working with that method.  I'll keep an eye out for upstream to package it all together.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10994#issuecomment-484958698

Thank you

File: srcpkgs/minitube/template [hunk 2]
```diff
@@ -1,17 +1,36 @@
 # Template file for 'minitube'
 pkgname=minitube
-version=2.9
+version=3.0
 revision=1
+_commit_hash_http="e790e31e6843c6ee8e97cd7e41e9eb9e024a9e2c"
```
Comment (maxice8) [inline srcpkgs/minitube/template]: doesn't need to be quoted

File: srcpkgs/minitube/template [hunk 3]
```diff
@@ -1,17 +1,36 @@
...
+ 98c8fb79044fe90bf92ee08f6f00ca2a68e17f36c0881ebb591e182e4b1d4f02
+ 597156e1f7ac3b8d8bdd251008cc0d10487c120462576088eed69425048d4bde"
+
+post_extract() {
+	mv -T "../http-${_commit_hash_http}" ${wrksrc}/lib/http
+	mv -T "../media-${_commit_hash_media}" ${wrksrc}/lib/media
+	mv -T "../idle-${_commit_hash_idle}" ${wrksrc}/lib/idle
+}
+
+
```
Comment (maxice8) [inline srcpkgs/minitube/template]: some extra lines here

File: srcpkgs/minitube/template (hunk 2 above)
Comment (ArmedAviator) [reply srcpkgs/minitube/template]: Fixed.

File: srcpkgs/minitube/template (hunk 3 above)
Comment (ArmedAviator) [reply srcpkgs/minitube/template]: Fixed.

**Pull request closed without merging**